scripts/protocgen.sh

```diff
@@ -13,22 +13,32 @@ protoc_gen_gocosmos() {
 
 protoc_gen_gocosmos
 
-proto_dirs=$(find ./proto -path -prune -o -name '*.proto' -print0 | xargs -0 -n1 dirname | sort | uniq)
-for dir in $proto_dirs; do
-  buf protoc \
-    -I "proto" \
-    -I "third_party/proto" \
-    --gocosmos_out=plugins=grpc,\
-Mgoogle/protobuf/any.proto=github.com/cosmos/cosmos-sdk/codec/types:. \
-    --grpc-gateway_out=logtostderr=true,allow_colon_final_segments=true:. \
-  $(find "${dir}" -maxdepth 1 -name '*.proto')
+protoc_gen_gopulsar() {
+  go install github.com/cosmos/cosmos-proto/cmd/protoc-gen-go-pulsar@latest 2>/dev/null
+}
 
+protoc_gen_gopulsar
+
+echo "Generating gogo proto code"
+cd proto
+proto_dirs=$(find ./cosmos -path -prune -o -name '*.proto' -print0 | xargs -0 -n1 dirname | sort | uniq)
+for dir in $proto_dirs; do
+  for file in $(find "${dir}" -maxdepth 1 -name '*.proto'); do
```

Comment:
do you need this for loop? for reference in tendermint we do https://github.com/tendermint/tendermint/blob/master/scripts/protocgen.sh#L32

Reply:
it is needed because we _only_ generate proto files with a `go_package` option explicitly specified with gogo. if no go_package is set we assume it is pulsar only